opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -479,6 +475,50 @@ func resourceOpennebulaVirtualMachineUpdate(d *schema.ResourceData, meta interfa
 		}
 	}
 
+	if d.HasChange("nic") {
+
+		log.Printf("[INFO] Update NIC configuration")
+
+		old, new := d.GetChange("nic")
+		attachedNicsCfg := old.([]interface{})
+		newNicsCfg := new.([]interface{})
+
+		timeout := d.Get("timeout").(int)
+
+		// get the list of nics ID to detach
+		toDetach := diffIDsConfig(attachedNicsCfg, newNicsCfg, "network_id")
+
+		// Detach the nics
+		for _, nicConfig := range toDetach {
+
+			nicID := nicConfig["nic_id"].(int)
+
+			err := vmNICDetach(vmc, timeout, nicID)
+			if err != nil {
+				return fmt.Errorf("vm disk detach: %s", err)
+
+			}
+		}
+
+		// get the list of nics to attach
+		toAttach := diffIDsConfig(newNicsCfg, attachedNicsCfg, "network_id")
+
+		// Attach the disks
+		for _, nicConfig := range toAttach {
+
+			networkID := nicConfig["network_id"].(int)
+			diskTpl := makeNICVector(map[string]interface{}{
+				"network_id": networkID,
+				"ip":         nicConfig["ip"],
+			})
```

Comment:
This is not enough compare to what it is possible to do when a NIC is attached during VM creation.
You must be able to do the same.
Here is the piece of code from shared_schemas.go:
```
    for i := 0; i < len(nics); i++ {                                                                    
        nicconfig := nics[i].(map[string]interface{})                                                   
        nic := tpl.AddNIC()                                                                             
                                                                                                        
        for k, v := range nicconfig {                                                                   
                                                                                                        
            if k == "network_id" {                                                                      
                nic.Add(shared.NetworkID, strconv.Itoa(v.(int)))                                        
                continue                                                                                
            }                                                                                           
                                                                                                        
            if isEmptyValue(reflect.ValueOf(v)) {                                                       
                continue                                                                                
            }                                                                                           
                                                                                                        
            switch k {                                                                                  
            case "ip":                                                                                  
                nic.Add(shared.IP, v.(string))                                                          
            case "mac":                                                                                 
                nic.Add(shared.MAC, v.(string))                                                         
            case "model":                                                                               
                nic.Add(shared.Model, v.(string))                                                       
            case "physical_device":                                                                     
                nic.Add("PHYDEV", v.(string))                                                           
            case "security_groups":                                                                     
                nicsecgroups := ArrayToString(v.([]interface{}), ",")                                   
                nic.Add(shared.SecurityGroups, nicsecgroups)                                            
            }                                                                                           
        }   
```